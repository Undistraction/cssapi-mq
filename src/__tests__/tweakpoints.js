import cssSerialiser from './helpers/cssSerialiser';
import {
  mqWithValidBreakpointsForRange,
  mqWithTweakedBreakpointsForRange,
} from './data';

expect.addSnapshotSerializer(cssSerialiser);

describe('tweak()', () => {
  it('throws if no breakpoints are supplied', () => {
    expect(() =>
      mqWithValidBreakpointsForRange('width').tweak()
    ).toThrowErrorMatchingSnapshot();
  });

  it('throws if no breakpoint sets are supplied', () => {
    expect(() =>
      mqWithValidBreakpointsForRange('width').tweak({})
    ).toThrowErrorMatchingSnapshot();
  });

  it('throws if invalid breakpoint value is supplied', () => {
    expect(() =>
      mqWithValidBreakpointsForRange('width').tweak({ width: { small: 'xxx' } })
    ).toThrowErrorMatchingSnapshot();
  });
});

// -----------------------------------------------------------------------------
// Tweaked
// -----------------------------------------------------------------------------

describe('tweaked()', () => {
  it('includes original breakpoints and added tweakpoints', () => {
    expect(
      mqWithTweakedBreakpointsForRange('width').tweaked.aboveWidth('alpha')`
      background-color: ${() => 'GhostWhite'};
    `
    ).toMatchSnapshot();

    expect(
      mqWithTweakedBreakpointsForRange('width').tweaked.betweenWidths(
        'alpha',
        'large'
      )`
      background-color: ${() => 'GhostWhite'};
    `
    ).toMatchSnapshot();
  });

  it("doesn't effect the original mq", () => {
    expect(
      () =>
        mqWithTweakedBreakpointsForRange('width').aboveWidth('alpha')`
      background-color: ${() => 'GhostWhite'};
    `
    ).toThrowErrorMatchingSnapshot();

    // Make sure the upper limit is 'medium', not 'alpha'
    expect(
      mqWithTweakedBreakpointsForRange('width').tweaked.atWidthBreakpoint(
        'small'
      )`
      background-color: ${() => 'GhostWhite'};
    `
    ).toMatchSnapshot();
  });
});
