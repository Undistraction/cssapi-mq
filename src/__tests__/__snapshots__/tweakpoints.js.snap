// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tweak() throws if invalid breakpoint value is supplied 1`] = `You must supply a 'dimension' as either a unitless positive number or a string comprised of a number followed by em, rem px but you supplied ["xxx"]`;

exports[`tweak() throws if no breakpoint sets are supplied 1`] = `You must supply at least one set of breakpoints to 'configure()', but the you supplied '{}'.`;

exports[`tweak() throws if no breakpoints are supplied 1`] = `You must supply at least one set of breakpoints to 'configure()', but the you supplied 'undefined'.`;

exports[`tweaked() doesn't effect the original mq 1`] = `There is no 'width' breakpoint defined called 'alpha', only: 'small, medium, large, xLarge' are defined.`;

exports[`tweaked() doesn't effect the original mq 2`] = `@media screen and (min-width: 25em) and (max-width: 56.249375em)`;

exports[`tweaked() includes original breakpoints and added tweakpoints 1`] = `@media screen and (min-width: 18.75em)`;

exports[`tweaked() includes original breakpoints and added tweakpoints 2`] = `@media screen and (min-width: 68.75em) and (max-width: 18.749375em)`;
